### PR TITLE
JavaScript: Make JSON and YAML dbscheme snippets more self-contained.

### PR DESCRIPTION
**Pull request by xiemaisi**
The JSON and YAML extractors are fairly independent of the JavaScript extractor and mostly work within their own part of the dbscheme and the standardised bits. There previously were only two exceptions to this: they both used the `filetype` relation to record (you guessed it) file types, and the `hasLocation` relation to associate JSON and YAML values, respectively, with their locations.

This PR removes the latter exception: the JSON extractor now puts locations into `json_locations` and the YAML extractor into `yaml_locations`. This difference is hidden by the library, so users of the standard JavaScript library are not affected by this change and old snapshots can be upgraded with full compatibility.

The advantage of this change is that other languages can now add the relevant snippets to their dbscheme (including the `filetype` relation if they don't have it) and then add JSON/YAML data to their snapshots using the existing extractors without them relying on an existing `hasLocation` relation.

I have verified that this does not affect performance or results on `nightly.slugs`, and I have manually tested upgrading a snapshot to ensure that locations for JSON and YAML elements are preserved.

I have also snuck in one more change to the dbscheme, adding relations used by the properties extractor (cf https://github.com/Semmle/ql/pull/473). While I have no immediate plans to use this, at least one prospective customer has asked about it, so I thought I might as well add it while I was working on the dbscheme anyway.

### CODE REVIEW

File: javascript/ql/src/semmlecode.javascript.dbscheme [hunk 1]
```diff
@@ -915,20 +918,22 @@ case @json_value.kind of
...
+    | @js_parse_error | @regexp_parse_error
     | @regexpterm
-    | @json_value
+    | @json_locatable
     | @token
     | @cfg_node
     | @jsdoc | @jsdoc_type_expr | @jsdoc_tag
-    | @yaml_node | @yaml_error
+    | @yaml_locatable
     | @xmllocatable;
```
Comment (asger-semmle) [inline javascript/ql/src/semmlecode.javascript.dbscheme]: Might as well include `@configLocatable` now that you included it in the dbscheme.

Comment (xiemaisi) [reply javascript/ql/src/semmlecode.javascript.dbscheme]: Good idea.

Comment (xiemaisi) [reply javascript/ql/src/semmlecode.javascript.dbscheme]: Done.

Comment (xiemaisi) [reply javascript/ql/src/semmlecode.javascript.dbscheme]: Done.

**Pull request merged**